manual/manual/cmds/intf-c.etex

```diff
@@ -989,14 +989,33 @@ invalidate the first argument after it is computed.
 Use the normal C array syntax instead.
 
 \begin{gcrule} Global variables containing values must be registered
-with the garbage collector using the "caml_register_global_root" function.
+with the garbage collector using the "caml_register_global_root" function,
+save that global variables and locations that will only ever contain OCaml
+integers (and never pointers) do not have to be registered.
+
+The same is true for any memory location outside the OCaml heap that contains a
+value and is not guaranteed to be reachable---for as long as it contains such
+value---from either another registered global variable or location, local
+variable declared with "CAMLlocal" or function parameter declared with
+"CAMLparam".
 \end{gcrule}
```

Comment:
That's a better description than the original, but I think it has too many exceptions :-)  What about:

"Global variables containing values must be registered
with the garbage collector using the "caml_register_global_root" function.
The same is true for any memory location outside the OCaml heap that contains a
value.  As an exception, registration is not necessary if ... or .... ".

Reply:
I don't agree with the "is not guaranteed to be reachable" clause. Even if the value is reachable from another location, it may be moved by the GC, which will make your pointer invalid.